### PR TITLE
Implement WallPost functionallity and Follows functionallity

### PR DESCRIPTION
**Pull request by kndynow**
**WallPost**

- Implemented `WallPostService` with logic to:
  - Create posts (validates content and that author/target user exist).
  - Get wall posts for a specific target user.
  - Get all posts by a given author.
  - Get a feed based on which users you follow.
  - Update and delete posts.
- Added API endpoints in `WallPostEndpoints`:
  - `POST /api/WallPosts/create`
  - `GET /api/WallPosts/target/{targetUserId}`
  - `GET /api/WallPosts/author/{authorId}`
  - `GET /api/WallPosts/feed/{userId}`
  - `PATCH /api/WallPosts/update`
  - `DELETE /api/WallPosts/delete/{id}`

**Follow**

- Added/extended the `Follow` entity with relationships between `Follower` and `FollowedUser` plus `CreatedAt`.
- Implemented `FollowService` with logic to:
  - Unfollow a user (`UnfollowAsync`) with checks for valid IDs and that the relationship exists.
  - Get followers (`GetFollowersAsync`) and followed users (`GetFollowedUsersAsync`) with validation that the user exists.
- Added API endpoints in `FollowEndpoints`:
  - `POST /api/Follows/follow`
  - `POST /api/Follows/unfollow`
  - `GET /api/Follows/followers/{userId}`
  - `GET /api/Follows/followed/{userId}`

### CODE REVIEW

File: backend/src/FriendZonePlus.Application/Services/FollowService.cs [hunk 1]
```diff
@@ -1,20 +1,27 @@
...
+        IFollowValidator validator,
+        IUserRepository userRepository)
     {
         _followRepository = followRepository;
         _validator = validator;
+        _userRepository = userRepository;
     }
 
-    public async Task FollowAsync(int followerId, int followedUserId)
+    //TODO: Test edge cases
```
Comment (AiWictoria) [inline backend/src/FriendZonePlus.Application/Services/FollowService.cs]: This can be removed as well since this to do is already done

**Review by AiWictoria — CHANGES_REQUESTED** [review]
Well done with mapping to DTO! Remember to remove TODO comments and dont use comments to comment what method names already says :)

**Review by AiWictoria — APPROVED** [review]
